grafana-plugin/src/containers/ColumnsSelectorWrapper/ColumnsSelectorWrapper.tsx

```diff
@@ -125,7 +131,10 @@ const ColumnsSelectorWrapper: React.FC<ColumnsSelectorWrapperProps> = observer((
   }
 
   async function onColumnRemovalClick(): Promise<void> {
-    const columns = store.alertGroupStore.columns.filter((col) => col.id !== columnToBeRemoved.id);
+    const columns = store.alertGroupStore.columns.filter(
+      (col) =>
+        col.id !== columnToBeRemoved.id || (col.id === columnToBeRemoved.id && col.type !== columnToBeRemoved.type)
```

Comment:
Can't we achieve the same with `({ id, type }) => id !== columnToBeRemoved.id && type !== columnToBeRemoved.type`? 🤔

Reply:
No because it's not the same condition check. Yours assumes that the id always needs to be different, which is not the case.